leaflet-treering.js

```diff
@@ -221,43 +228,47 @@ function MeasurementData (dataObject) {
   * Add a new point into the measurement data
   * @function newPoint
   */
-  MeasurementData.prototype.newPoint = function(start, latLng, hasLatewood) {
+  MeasurementData.prototype.newPoint = function(start, latLng, subAnnual, forwardDirection) {
     if (start) {
       this.points[this.index] = {'start': true, 'skip': false, 'break': false, 'latLng': latLng};
     } else {
       this.points[this.index] = {'start': false, 'skip': false, 'break': false, 'year': this.year, 'earlywood': this.earlywood, 'latLng': latLng};
-      if (hasLatewood) {
+      if (subAnnual) { // check if points alternate ew & lw
         if (this.earlywood) {
           this.earlywood = false;
         } else {
           this.earlywood = true;
-          this.year++;
+          if (forwardDirection) {
```

Comment:
I don't like forwardDirection as a name and concept, because it makes the else{} conditions confusing.

My instinct would be to define two constant values storing an actual direction and compare with those - that makes it much more understandable.

const FORWARD_MEASUREMENT = "forward";
const REVERSE_MEASUREMEN = "reverse"; (whatever names make sense0

if(direction == FORWARD_MEASUREMENT) {
code
}
else if(direction == REVERSE_MEASUREMENT) {
code
}

Reply:
What I was thinking with forwardDirection is to keep it simple with true/false. forward refers to forward in time (1901 to 1902 etc.) If forwardDriection is false that implies backwards (1901 to 1900 etc. ). I can add comments to make that more clear.